File: src/001.Array/414.ThirdMaximumNumber/414.第三大的数.js
/*
 * @lc app=leetcode.cn id=414 lang=javascript
 *
 * [414] 第三大的数
 */

// @lc code=start
/**
 * @param {number[]} nums
 * @return {number}
 */
var thirdMax = function (nums) {
	const filterNums = Array.from(new Set(nums)).sort((a, b) => b - a);

	if (filterNums.length >= 3) {
		return filterNums[2];
	}

	return filterNums[0];
};
// console.log(thirdMax([2, 1]));
// @lc code=end

File: src/001.Array/414.ThirdMaximumNumber/414.第三大的数.go
/*
 * @lc app=leetcode.cn id=414 lang=golang
 *
 * [414] 第三大的数
 */

// @lc code=start
func thirdMax(nums []int) int {
	arr := uniqueArr(nums)
	sort.Sort(sort.Reverse(sort.IntSlice(arr)))

	if len(arr) >= 3 {
		return arr[2]
	}

	return arr[0]
}

// 数组去重
func uniqueArr(nums []int) []int {
	arr := make([]int, 0)
	tempMap := make(map[int]bool, len(nums))

	for _, v := range nums {
		if tempMap[v] == false {
			tempMap[v] = true
			arr = append(arr, v)
		}
	}

	return arr
}

// @lc code=end

